server/routes/induction/exemption/exemption/exemptionReasonController.test.ts

```diff
@@ -87,7 +88,7 @@ describe('exemptionReasonController', () => {
       // Then
       expect(res.redirect).toHaveBeenCalledWith(`/prisoners/A1234BC/induction/${journeyId}/exemption/confirm`)
       expect(getPrisonerContext(req.session, prisonNumber).inductionExemptionForm).toBeUndefined()
-      expect(getPrisonerContext(req.session, prisonNumber).inductionExemptionDto).toEqual(inductionExemptionDto)
+      expect(req.journeyData.inductionExemptionDto).toEqual(inductionExemptionDto)
```

Comment:
Worth noting that this PR is only about moving the DTO (the object that is passed from page to page) from prisonerContext/session to the new journeyData.

The individual page forms are still held on the prisonerContext/session - these are less of a problem for what we are trying to solve, but they will get moved/refactored, but will involve some very different refactoring so I wanted to keep them separate.